lib/src/lifecycle_module.dart

```diff
@@ -801,6 +801,7 @@ abstract class LifecycleModule extends SimpleModule
       rethrow;
     } catch (error, stackTrace) {
       _didUnloadController.addError(error, stackTrace);
+      await _disposableProxy.dispose();
```

Comment:
Could wrap this call in a nested `try` block and then put lines 805 and 806 in its `finally` clause...?